### DISCUSSION TITLE
Error: Event blockUpdate:(-12, -60, 3)

### DISCUSSION THREAD

**Discussion by Vyakoo** [PrismarineJS/mineflayer#3208] [Q&A]
I saw a similar post, but this post does not solve the problem.

### Error: Event blockUpdate:(-12, -60, 3) did not fire within timeout of 5000ms
    at onceWithCleanup (A:\My project\JavaScript\mineflayer\test\node_modules\mineflayer\lib\promise_utils.js:62:26)
    at placeBlockWithOptions (A:\My project\JavaScript\mineflayer\test\node_modules\mineflayer\lib\plugins\place_block.js:13:36)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async EventEmitter.placeBlock (A:\My project\JavaScript\mineflayer\test\node_modules\mineflayer\lib\plugins\place_block.js:33:5)
    at async Farm.startFarming (A:\My project\JavaScript\mineflayer\test\farm.js:67:15)

**Comment by amoraschi** [comment]
We cannot provide help if you don't provide the code you're using/the other post you're talking about/the full error

**Comment by Vyakoo** [comment]
Sorry
`Error: Event blockUpdate:(-12, -60, 3) did not fire within timeout of 5000ms
    at onceWithCleanup (A:\My project\JavaScript\mineflayer\test\node_modules\mineflayer\lib\promise_utils.js:62:26)
    at placeBlockWithOptions (A:\My project\JavaScript\mineflayer\test\node_modules\mineflayer\lib\plugins\place_block.js:13:36)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async EventEmitter.placeBlock (A:\My project\JavaScript\mineflayer\test\node_modules\mineflayer\lib\plugins\place_block.js:33:5)
    at async Farm.startFarming (A:\My project\JavaScript\mineflayer\test\farm.js:67:15)`


My code is based on the example https://github.com/PrismarineJS/mineflayer/blob/master/examples/farmer.js


### const {pathfinder, Movements, goals : {GoalNear, GoalFollow}} = require('mineflayer-pathfinder');
const { Vec3 } = require('vec3')



module.exports = class Farm {
    static bot;
    static mcData;
    static Timeout = setTimeout(() => this.startFarming, 1000);



    /**
     * 
     * @param {import('mineflayer').this.bot} this.bot 
     * @param {import('minecraft-data')} this.mcData 
     */

    constructor(bot, mcData) {
        this.bot = bot;
        this.mcData = mcData;
    }


    blockPlace() {
        return this.bot.findBlock({
          point: this.bot.entity.position,
          maxDistance: 20,
          matching: this.bot.registry.blocksByName.farmland.id,
          useExtraInfo: (block) => {
            const blockAbove = this.bot.blockAt(block.position.offset(0, 1, 0));
            return blockAbove.type === 0
          }
        });
      }
      
      
      blockHarvest() {
        return this.bot.findBlock({
          point: this.bot.entity.position,
          maxDistance: 20,
          matching: (block) => {
            return block && block.type === this.bot.registry.blocksByName.wheat.id && block.metadata === 7
          }
        });
      }
      
      
      async startFarming() {
      
        try {
          while(1) {
            const toHarvest = this.blockHarvest();
      
            if(toHarvest) {

              try {
                this.bot.pathfinder.setMovements(new Movements(this.bot, this.mcData));
                this.bot.pathfinder.setGoal(new GoalNear(toHarvest.position.x, toHarvest.position.y, toHarvest.position.z, 1));
              } catch (error) {
                console.log(error)
              }

              
              this.bot.once(`goal_reached`, async () => {
                await this.bot.dig(toHarvest);
              });
              
            } else {
              break
            }
          }
      
          while(1) {
            const toPlace = this.blockPlace();
      
            if(toPlace) {
              await this.bot.equip(this.bot.registry.itemsByName.wheat_seeds.id, 'hand');

              this.bot.pathfinder.setMovements(new Movements(this.bot, this.mcData));
              this.bot.pathfinder.setGoal(new GoalNear(toPlace.position.x, toPlace.position.y, toPlace.position.z, 1));

              this.bot.once(`goal_reached`, async () => {
                await this.bot.placeBlock(toPlace, new Vec3(0, 1, 0));
              });

            } else {
              break
            }
          }
        } catch (error) {
          console.log(error)
        }
      
        this.Timeout
      
      }
      
      stopFarming() {
        clearTimeout(Timeout);
      }

}
